sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/planner/BeamRuleSets.java

```diff
@@ -42,11 +49,18 @@
 @Internal
 public class BeamRuleSets {
 
-  public static RuleSet[] getRuleSets() {
-    return new RuleSet[] {
-      RuleSets.ofList(
-          BeamProjectRule.INSTANCE,
-          BeamFilterRule.INSTANCE,
+  public static List<RelOptRule> LOGICAL_OPTIMIZATIONS =
+      ImmutableList.of(
+          // Rules so we only have to implement Calc
```

Comment:
I believe FilterToCalcRule and ProjectToCalcRule eliminate all filters and projects. But it is very hard to know with Calcite whether rules are sufficient. I got no answer on their mailing list so it might just be we need a huge test suite.